### DISCUSSION TITLE
How to noqa E501 in docstring?

### DISCUSSION THREAD

**Discussion by jamesbraza** [astral-sh/ruff#7011] [Q&A] [answered]
Here is some Python code with a Google-style docstring, containing a Mermaid diagram rendered by Sphinx:

```python
def foo() -> None:
    """
    Some Google-style docstring.

    # ruff: noqa: E501

    .. mermaid::

       flowchart TD
           A([Start]) --> B["`Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples`"]
    """
```

The last line is designed to trigger `E501`.  Currently with `ruff==0.0.286`:

```bash
> ruff --isolated --select=E501 --line-length=88 a.py
a.py:10:98: E501 Line too long (109 > 88 characters)
Found 1 error.
``` 

How can I E501:
- Just this docstring
- Without adding a comment to the `flowchart` source line?

**Comment by charliermarsh** [comment]
Can you try adding the noqa just after the closing triple-quotes?

**Reply by jamesbraza** [reply]
```python
def foo() -> None:
    """  # ruff: noqa: E501
    Some Google-style docstring.

    .. mermaid::

       flowchart TD
           A([Start]) --> B["`Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples`"]
    """  # ruff: noqa: E501
```

Trying either end of the triple quotes doesn't properly disable the `E501`, and using the 2nd triple quote also leads to a warning:

```bash
> ruff --isolated --select=E501 --line-length=88 a.py
warning: Unexpected `# ruff: noqa` directive at a.py:9. File-level suppression comments must appear on their own line.
a.py:8:98: E501 Line too long (109 > 88 characters)
Found 1 error.
```

**Reply by zanieb** [reply, marked as answer]
If you do

```python
def foo() -> None:
    """
    Some Google-style docstring.

    .. mermaid::

       flowchart TD
           A([Start]) --> B["`Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples Apples`"]
    """  # noqa: E501
```

it will work. `ruff: noqa` is for file-level suppression and shouldn't be used for suppressing a single item — which is why you get that warning.

**Reply by jamesbraza** [reply]
Ah that is what I needed, thank you!

I think this `# noqa` _after_ the docstring is worth adding to the docs: https://beta.ruff.rs/docs/tutorial/#ignoring-errors

I can contribute something

**Reply by charliermarsh** [reply]
It’s mentioned here, but not in the tutorial: https://beta.ruff.rs/docs/configuration/#error-suppression

**Reply by jamesbraza** [reply]
Oh I opened a [PR](https://github.com/astral-sh/ruff/pull/7014), but seeing that section, I can make it better in the morning.

**Reply by jamesbraza** [reply]
Alright, updated the PR to link the tutorial to the error suppression docs 👌
